src/proc/colorizer.cpp

```diff
@@ -235,31 +235,13 @@ namespace librealsense
 
     rs2::frame colorizer::process_frame(const rs2::frame_source& source, const rs2::frame& f)
     {
+        float depth_units = -1;
         if (f.get_profile().get() != _source_stream_profile.get())
         {
             _source_stream_profile = f.get_profile();
             _target_stream_profile = f.get_profile().clone(RS2_STREAM_DEPTH, f.get_profile().stream_index(), RS2_FORMAT_RGB8);
-
-            auto snr = ( (frame_interface *)f.get() )->get_sensor().get();
-            auto depth_sensor = As< librealsense::depth_sensor >( snr );
-            if( depth_sensor )
-                _depth_units = depth_sensor->get_depth_scale();
-            else
-            {
-                // For playback sensors
-                auto extendable = As< librealsense::extendable_interface >( snr );
-                if( extendable
-                    && extendable->extend_to( TypeToExtension< librealsense::depth_sensor >::value,
-                                              (void **)( &depth_sensor ) ) )
-                {
-                    _depth_units = depth_sensor->get_depth_scale();
-                }
-                else
-                {
-                    LOG_ERROR( "Failed to query depth units from sensor" );
-                    throw std::runtime_error( "failed to query depth units from sensor" );
-                }
-            }
+            depth_units = ((depth_frame*)f.get())->get_units();
+            
```

Comment:
@ev-mp  I still think this part is redundant because depth_units is defined the same way regardless to playback condition, but I will add it for you to see